site/source/tutorials/1040_data_driven_widgets/index.md

```diff
@@ -0,0 +1,193 @@
+---
+layout: tutorial
+title: Data-driven widgets
+overview: In this tutorial, you will learn how to create reactive data-driven widgets in Dojo 2.
+---
+
+{% section 'first' %}   
+
+# Data-driven widgets
+
+## Overview
+The Dojo 2 widget system provides a functional API that attempts to strictly enforce a unidirectional data flow: the only way to interact with a widget is through the `properties` it exposes, and dealing directly with widget instances is both uncommon and bad practice. It can be confusing to understand how to build data-driven widgets in such a reactive framework, especially when widgets in past frameworks like Dojo 1 were so tightly coupled to store implementations.
+
+In this tutorial, a filterable data-driven list widget will be built to demonstrate how Dojo 2 widgets should be decoupled from data providers.
+
+## Prerequisites
+You can [download](../assets/006_data_driven_widgets-initial.zip) the demo project and run `npm install` to get started.
+
+The `@dojo/cli` command line tool should be installed globally. Refer to the [Dojo 2 local installation](../000_local_installation/) article for more information.
+
+You also need to be familiar with TypeScript as Dojo 2 uses it extensively. For more information, refer to the [TypeScript and Dojo 2](../comingsoon.html) article.
+
+{% section %}
+
+## Creating a filterable List widget
+
+{% task 'Create a widget that uses a Dojo 2 `TextInput`.' %}
+
+Before digging into the specifics of wiring a widget to a data source, a basic list widget with a filter input must first be created. For now, the widget will only render a Dojo 2 `TextInput`, but functionality will be added throughout the tutorial. The implementation will be put in `List.ts`, and just like with other widgets created during previous tutorials, initial dependencies and a class declaration are needed to get started.
```

Comment:
The implementation will be put in `List.ts`, -> `List.ts` will contain our implementation,